### PR TITLE
chore: prepare for docs versioning

### LINKED ISSUE #6548: Add versioning support to documentation website
- do we wanna look into adding versioning, or at least show the lasted version here?
image

_Originally posted by @nflaig in https://github.com/ChainSafe/lodestar/issues/6533#issuecomment-1997902396_

### PR DESCRIPTION
**Pull request by jeluard**
**Motivation**

Prepare for future versioning. For now only support one version, and display it in the main toolbar.

Capture d’écran 2024-03-18 à 13 58 10

Fixes #6548

### CODE REVIEW

File: docs/docusaurus.config.ts [hunk 1]
```diff
@@ -37,6 +37,14 @@ const config: Config = {
           sidebarPath: "./sidebars.ts",
           editUrl: "https://github.com/ChainSafe/lodestar/tree/unstable/docs/",
           routeBasePath: "/",
+          lastVersion: 'current',
+          versions: {
+            current: {
+              label: '1.7.0'
```
Comment (wemeetagain) [inline docs/docusaurus.config.ts]: shouldn't this match whats in our packages/cli/package.json?
What is the process for synchronizing this?

**Comment by jeluard** [comment]
My bad, my intention was to match the repo versions. Ideally the latest version would be updated via the release scripts.

It's not clear to me when we would want to introduce a new doc version. FWIW, introducing a new doc version requires duplicating the whole doc content in a specific repo (a script can help this process)

Maybe @nflaig can shim in, as was interested with the versioning support.

**Comment by nflaig** [comment]
> Maybe @nflaig can shim in, as was interested with the versioning support.

The idea was to have something similar to what teku does https://docs.teku.consensys.io/.

We could always build docs from unstable branch (on each commit), have docs from stable branch as default as we do now, and then just a few previous releases back.

>  FWIW, introducing a new doc version requires duplicating the whole doc content in a specific repo (a script can help this process)

Previous docs bundle would have to be hosted somewhere, maybe this is possible via github pages, or another source is required. Might wanna check out how teku does it https://github.com/Consensys/doc.teku

**Comment by jeluard** [comment]
> Previous docs bundle would have to be hosted somewhere, maybe this is possible via github pages, or another source is required. Might wanna check out how teku does it https://github.com/Consensys/doc.teku

Sorry I meant in a specific folder, it's fine in current repo/gh-pages hosting. So the repo would ultimately contain multiple copies of the whole documentation, with just a couple changes (see how it looks like for [teku](https://github.com/Consensys/doc.teku/tree/main/versioned_docs)).

Maybe we can decide to not introduce doc version per lodestar release, but only introduce new doc version when significant docs changes have been introduced.

File: docs/versions.json [hunk 2]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (nflaig) [inline docs/versions.json:1]:
```suggestion
[]

```

**Review by nflaig — COMMENTED** [review]
Will this just show the latest version for now?

**Comment by jeluard** [comment]
Closing for now. Will address once we have a better idea of what we want.

**Comment by nflaig** [comment]
> Closing for now. Will address once we have a better idea of what we want.

The ideal case would be the same as Teku docs, have unstable / stable, and a few releases back

image

But if that's too much effort to realize it's definitely not that high prio, so can postpone that for now

**Pull request closed without merging**